### DISCUSSION TITLE
Add syntax highlighting for HTML entities

### DISCUSSION THREAD

**Discussion by sdalmonte** [zed-industries/zed#23927] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

HTML [entities](https://developer.mozilla.org/en-US/docs/Glossary/Entity) are currently treated as text by syntax highlighting. This could incorrectly imply that entities are rendered as plain text. A distinct colour of syntax highlighting could indicate that entities will be translated into their corresponding character.

i.e. `1920&times;1080, 2023&ndash;2024, Zed&reg;` → 1920×1080, 2023–2024, Zed®

**Comment by cole-miller** [comment]
From the Zed team: this seems like it might be a reasonable feature! If anyone wants to take a crack at an implementation, feel free to send a PR. In the meantime, I'm converting this issue to a discussion since that's a more appropriate setting for hashing out details like what languages should have this highlighting, etc.
